### PR TITLE
Feature/details component

### LINKED ISSUE #16: ForecastDetails Container component
Create component to display 5 day forecast details when user clicks on home city card

### LINKED ISSUE #17: ForecastDetails Card
Individual card element - each card will be a new day of forecast for specific home city card location (4-5 forecast details cards will be held in forecast details container, 5-day forecast)

### LINKED ISSUE #18: TodayForecast Details component
Create component to hold a more detailed forecast for today's weather. Once user clicks on home city card, TodayForecastDetails component will be above ForecastDetails Container (5-day forecast)

### LINKED ISSUE #19: Interface for TodayForecastDetails
Create interface types for TodaysForecastDetails component

### LINKED ISSUE #20: Interface - ForecastDetails Container & ForecastDetail Cards
Create interface for ForecastDetails container and cards

### LINKED ISSUE #24: Set up Router / Links for home page and details pages
When user clicks on HomeCityCard, they are taken to a new page that displays a more details weather forecast for today (TodaysForecastDetails) plus a 5 day forecast (ForecastDetails component). Set up Router/Switch/Links in order to add this feature.

### PR DESCRIPTION
**Pull request by nataliemcintyre2021**
# Description

- Router added and city cards assigned link with city name
- On click user is taken to a details page for city weather
- Think about implementing Switch statements and fix issue of URL showing not on click

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] Tested via React Dev Tools and console.log

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Components/App/App.tsx [hunk 1]
```diff
@@ -44,23 +46,42 @@ const App: FC = () => {
...
+              setDetails(city)
+            }}
+            clickedCard={currentCity}
+            details={forecastDetails}
+          />}
+        </main>
+        </>
+      )
+    }} />
+    <Route exact path="/:name" render={({ match }) => {
```
Comment (e-spitz) [inline src/Components/App/App.tsx:73]: Might need to edit this Route path to perform another fetch call so the localhost://3000/CityName will actually take them to a forecast page and not an empty page.

Comment (darlaevans2000) [inline src/Components/App/App.tsx:73]: This route works perfectly but if we just do localhost:3000/cityname we run into an error now.

File: src/Components/HomeCard/HomeCard.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { WeatherLocation, Forecast } from "../../model/Weather";
 interface HomeCardProps {
   key: number;
   onSelect: (city: WeatherLocation) => void;
-  current: WeatherLocation;
+  clickedCard: WeatherLocation;
```
Comment (e-spitz) [inline src/Components/HomeCard/HomeCard.tsx:8]: This is much more clear for me to understand!

File: src/Components/HomeCityCards/HomeCityCards.tsx [hunk 3]
```diff
@@ -3,30 +3,33 @@ import './HomeCityCards.css'
...
 import { WeatherLocation, Forecast } from "../../model/Weather";
 import HomeCard from "../HomeCard/HomeCard"
+import { Link } from 'react-router-dom';
 
 interface HomeCityProps {
   allCities: WeatherLocation[];
   onSelect: (city: WeatherLocation) => void;
-  current: WeatherLocation | null;
+  clickedCard: WeatherLocation | null;
   details: Forecast | null;
```
Comment (e-spitz) [inline src/Components/HomeCityCards/HomeCityCards.tsx:12]: Going to need to add a <Loading /> component for the delay in fetching the details and setting that to state.

**Review by e-spitz — APPROVED** [review]

File: src/Components/HomeCityCards/HomeCityCards.tsx [hunk 4]
```diff
@@ -3,30 +3,33 @@ import './HomeCityCards.css'
...
 }
 
-export const HomeCityCards: FC<HomeCityProps> = ({ allCities, onSelect, current, details }) => {
+export const HomeCityCards: FC<HomeCityProps> = ({ allCities, onSelect, clickedCard, details }) => {
 console.log('forecast details', details)
 
   const cityCards = allCities.map(city => {
```
Comment (darlaevans2000) [inline src/Components/HomeCityCards/HomeCityCards.tsx:18]: I love this destructuring!

**Review by darlaevans2000 — APPROVED** [review]

**Pull request merged**